pkg/virt-controller/watch/vm.go

```diff
@@ -544,6 +546,40 @@ func (c *VMController) handleDataVolumes(vm *virtv1.VirtualMachine, dataVolumes
 	return ready, nil
 }
 
+// areDataVolumesReady determines whether all DataVolumes specified for a VM
+// have been successfully provisioned and/or claimed, and are ready for consumption.
+// Note that DataVolumes in WaitForFirstConsumer phase are not regarded as ready.
+func (c *VMController) areDataVolumesReady(vm *virtv1.VirtualMachine) bool {
+	for _, template := range vm.Spec.DataVolumeTemplates {
```

Comment:
i'd only run this check if the VM's VMI is not phase == running. Due to hotplug we can provision storage while the VMI is running, and i think that would get confusing to see a VM that's currently active (running) being reported as provisioning. 



Also, It's technically possible for someone to delete a DataVolume while the VMI is running, and I don't know what that does to the DataVolume's phase. The underlying PVC won't be removed until the VMI's pod exits though, so it wouldn't strip it out from under the VMI.

Reply:
> i'd only run this check if the VM's VMI is not phase == running. Due to hotplug we can provision storage while the VMI is running, and i think that would get confusing to see a VM that's currently active (running) being reported as provisioning.

Done. 

> Also, It's technically possible for someone to delete a DataVolume while the VMI is running, and I don't know what that does to the DataVolume's phase. The underlying PVC won't be removed until the VMI's pod exits though, so it wouldn't strip it out from under the VMI.

Just tried that, and it does nothing to the DataVolume's phase. Consequently, it does nothing to the VM's status too - I believe this is the right behavior. If the VMI is stopped and the DataVolume deletes successfully, we'd be back to Provisioning status.